llettuce-docs/embedding_models.rst

```diff
@@ -0,0 +1,307 @@
+Embedding Models
+=================
+
+This page contains details about the embedding models used in the project.
+
+.. _embedding-models:
+
+Embedding Models Documentation
+------------------------------
+
+This section contains details about the embedding models used in the project.
+
+
+Available Models:
+~~~~~~~~~~~~~~~~~
+
+1. (Bidirectional Gated Encoder - Small)
+
+BGESMALL
+--------
+    - Model Name: Bidirectional Gated Encoder
+    
+    - Version: Small
+    
+    - Dimensions: 384
+    
+    - Description:
+        BGE-small has been designed for tasks such as sentence
+        embedding, semantic similarity, and information retrieval. 
+        
+    - Benefits:
+        BGE-small more efficient in terms of speed and memory,
+        while still being capable of generating high-quality
+        sentence or document embeddings.
+        
+    - Research Paper:
+        arXiv:2402.03216 
+
+
+
+2. Sentence-BERT MiniLM
+
+MINILM
+------
+    - Model Name: Sentence-BERT
+    
+    - Version: MiniLM
+    
+    - Dimensions: 384
+    
+    - Description:
+        MiniLM is a smaller, more efficient version of the 
+        Sentence-BERT (SBERT) model, specifically optimized 
+        for tasks such as semantic similarity, sentence 
+        embeddings, and information retrieval.
+        
+    - Benefits:
+        The MiniLM variant provides a good trade-off between 
+        computational efficiency and performance, making it
+        suitable for use cases where speed and resource 
+        limitations are important, without sacrificing 
+        too much accuracy.
+        
+    - Research Paper:
+        arXiv:1908.10084
+
+
+
+3. Generalizable T5 Retrieval - Base
+
+GTR_T5_BASE
+-----------
+    - Model Name: Generalizable T5 Retrieval
+    
+    - Version: Base
+    
+    - Dimensions: 768
+    
+    - Description:
+        GTR-T5 is a dense retrieval model using a dual encoder 
+        architecture for efficient semantic search and passage 
+        retrieval. It encodes queries and documents separately
+        into a shared embedding space, allowing fast and scalable
+        retrieval using a dot-product similarity. This model 
+        is based on T5 and optimized for generalization 
+        across diverse tasks.
+
+    - Benefits:
+        GTR-T5 is highly generalizable across different tasks, scalable
+        in size, and efficient for large-scale retrieval with precomputed 
+        document embeddings. 
+        
+        It leverages the T5 model for deep semantic understanding and 
+        ensures fast retrieval using Approximate Nearest Neighbor 
+        (ANN) search, making it both powerful and efficient 
+        for various retrieval tasks.
+        
+    - Research Paper:
+        ArXiv:abs/2112.07899
+
+
+
+4. Generalizable T5 Retrieval (Large)
```

Comment:
If some of the models are different sized versions of one another, maybe run them as subheadings?